### PR TITLE
Test that type alias impl trait happens in a submodule

### PR DESCRIPTION
**Pull request by spastorino**
r? @oli-obk 

Related to #86727

### CODE REVIEW

File: src/test/ui/type-alias-impl-trait/defining-use-submodule.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// run-pass
+
+#![feature(type_alias_impl_trait)]
+
+// test that the type alias impl trait defining use is in a submodule
+
+fn main() {
```
Comment (jackh726) [inline src/test/ui/type-alias-impl-trait/defining-use-submodule.rs:7]: I don't think this actually needed. Could just be a check-pass test?

Comment (spastorino) [reply src/test/ui/type-alias-impl-trait/defining-use-submodule.rs:7]: Yes, was just following the intention of the original test https://github.com/rust-lang/rust/pull/88172/files#diff-f0138a92e0c5bb0a679a72ff03ec1c30cf35ab5cc36f1143efdda3c98ad91bd3L14
But agreed with you that we could just make a check-pass test instead

Comment (spastorino) [reply src/test/ui/type-alias-impl-trait/defining-use-submodule.rs:9]: Done!

**Pull request merged**